cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/JsonOutputFormatter.cs

```diff
@@ -0,0 +1,127 @@
+    {
+        using var reader = new StreamReader(content);
+        var strContent = reader.ReadToEnd();
+        if (options is JsonOutputFormatterOptions jsonOptions && jsonOptions.OutputIndented)
+        {
+            var result = ProcessJson(strContent, jsonOptions.OutputIndented);
+            _ansiConsole.WriteLine(result);
+        }
+        else
+        {
+            _ansiConsole.WriteLine(strContent);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task WriteOutputAsync(Stream content, IOutputFormatterOptions options, CancellationToken cancellationToken = default)
+    {
+        string resultStr;
+
+        if (options is JsonOutputFormatterOptions jsonOptions && jsonOptions.OutputIndented)
+        {
+            using var result = await ProcessJsonAsync(content, jsonOptions.OutputIndented, cancellationToken);
+            using var r = new StreamReader(result);
+            resultStr = await r.ReadToEndAsync();
+        }
+        else
+        {
+            using var reader = new StreamReader(content);
+            resultStr = await reader.ReadToEndAsync();
+        }
+
+        _ansiConsole.WriteLine(resultStr);
+    }
+
+    /// <summary>
+    /// Given a JSON input string, returns a processed JSON string with optional indentation
+    /// </summary>
+    /// <param name="input">JSON input string</param>
+    /// <param name="indent">Whether to return indented output</param>
+    private static string ProcessJson(string input, bool indent = true)
+    {
+        var result = input;
+        try
+        {
+            var jsonDoc = JsonDocument.Parse(input);
+            result = JsonSerializer.Serialize(jsonDoc, options: new() { WriteIndented = indent });
+        }
+        catch (JsonException)
+        {
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    /// Given a JSON input stream, returns a processed JSON stream with optional indentation
+    /// </summary>
+    /// <param name="input">JSON input stream</param>
+    /// <param name="indent">Whether to return indented output</param>
+    /// <param name="cancellationToken">The cancellation token</param>
+    private static async Task<Stream> ProcessJsonAsync(Stream input, bool indent = true, CancellationToken cancellationToken = default)
+    {
+        var buffer = new byte[input.Length];
+        await input.ReadAsync(buffer, cancellationToken);
+
+        try
+        {
+            using var cachedStream = new MemoryStream(buffer);
+            var jsonDoc = await JsonDocument.ParseAsync(cachedStream, default, cancellationToken);
+
+            var outputStream = new MemoryStream();
+            await JsonSerializer.SerializeAsync<object>(outputStream, jsonDoc, cancellationToken: cancellationToken, options: new() { WriteIndented = indent });
+            outputStream.Position = 0;
+            return outputStream;
+        }
+        catch (JsonException)
+        {
+            return new MemoryStream(buffer);
```

Comment:
this should either log or throw

Reply:
sure, but the fact that the CLI was not able to indent something properly should be logged, at least at a warning level, so people understand something went wrong somewhere and can report it to us, or our telemetry if we set any.

Reply:
The logging will happen a lot when we have a JmesPath query. Every time we have an object e.g. `{"a": "foo", "b": "bar", "c": "baz"}` and a JmesPath query `a`, the result of the query will be a string `foo` which isn't valid JSON. The `Parse` call will fail with a `JsonException`. I don't think logging this when we know it will happen is very useful.

Reply:
I see, thanks for going over that scenario, I hadn't anticipated that, resolving.